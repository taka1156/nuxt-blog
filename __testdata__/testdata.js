const dummyImgUrl = 'http://placehold.jp/150x150.png';
const dummyUrl = 'http://placehold.jp';
const dummyCopyrightUrl = 'http://placehold.jp';
const MAX_NAV_DATA = 4;
const MAX_ICONS_DATA = 4;
const MAX_ARTICLE_DATA = 10;
const MAX_BADGE_DATA = 5;
const MAX_TOC_DATA = 5;

const dummyOutsideLink = { routeTo: dummyUrl };
const dummyRouteStr = { routeTo: '/' };
const dummyRouteObj = {
  routeTo: {
    name: 'sample-id',
    params: { id: 1 }
  }
};

// factory
const dummyFactory = (num, fn) => {
  return [...new Array(num)].map((_, i) => fn(i + 1));
};

// template
const dummyNavTemplate = () => ({
  name: 'ダミーナビゲーション',
  to: '/test1',
  img: dummyImgUrl
});

const dummyBadgeTemplate = (i, name) => ({
  id: i,
  name: name,
  img: { url: dummyImgUrl }
});

const dummyClassificationTemplate = (i, name) => ({
  id: i,
  name: name,
  img: {
    url: dummyImgUrl
  }
});

const dummyTocTemplate = i => ({
  index: i,
  escapedText: 'この文章はダミーです。',
  anchor: `anchor_${i}`
});

const dummyMarkdown = `
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。** 
## この文章はダミーです。 
  **文字の大きさ、量、字間、行間等を確認するために入れています。** 
## この文章はダミーです。 
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
`;

const dummyArticleTemplate = i => ({
  id: i,
  ...dummyDate,
  title:
    'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています',
  summary:
    'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量',
  body: dummyMarkdown,
  tags: dummyTagBadges.slice(0, i),
  category: dummyBadgeTemplate(1, 'ダミーカテゴリー')
});

const dummySnsIconTemplate = () => ({
  name: 'ダミーアイコン',
  img: dummyImgUrl,
  link: dummyUrl
});

// dummyDate
const dummyImg = {
  imgUrl: dummyImgUrl,
  imgAlt: 'ダミーの画像'
};

const dummyDate = {
  createdAt: '2020-07-04T10:53:40.252Z',
  updatedAt: '2020-07-08T15:15:07.668Z'
};

const dummyRoutes = dummyFactory(MAX_NAV_DATA, () => dummyNavTemplate());

const dummyTagBadges = dummyFactory(MAX_BADGE_DATA, i =>
  dummyBadgeTemplate(i, 'ダミータグ')
);

const dummyClassificationCategory = dummyFactory(MAX_BADGE_DATA, i =>
  dummyClassificationTemplate(i, 'ダミーカテゴリー')
);

const dummyClassificationTag = dummyFactory(MAX_BADGE_DATA, i =>
  dummyClassificationTemplate(i, 'ダミータグ')
);

const dummyArticles = dummyFactory(MAX_ARTICLE_DATA, i => dummyArticleTemplate(i));

const dummyFactoryBadge = (id, name) => {
  const tmpTag = dummyBadgeTemplate(id, name);
  return {
    routePath: 'tag-id',
    badge: tmpTag
  };
};

const dummyCategoryBadge = dummyFactoryBadge(1, 'ダミーカテゴリー');

const dummyTagBadge = dummyFactoryBadge(1, 'ダミータグ');

const dummyPagination = {
  currentPage: 1,
  maxPage: Math.ceil(dummyArticles.length / 5)
};

const dummyTocs = dummyFactory(MAX_TOC_DATA, i => dummyTocTemplate(i));

const dummyLogo = 'ダミーロゴ';

const dummySnsIcons = dummyFactory(MAX_ICONS_DATA, () => dummySnsIconTemplate());

const dummyProfile = {
  img: dummyImgUrl,
  introduce:
    'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。'
};

export {
  dummyOutsideLink,
  dummyRouteStr,
  dummyRouteObj,
  dummyImg,
  dummyCategoryBadge,
  dummyTagBadge,
  dummyTagBadges,
  dummyDate,
  dummyArticles,
  dummyPagination,
  dummyClassificationCategory,
  dummyClassificationTag,
  dummyLogo,
  dummyTocs,
  dummyRoutes,
  dummyCopyrightUrl,
  dummySnsIcons,
  dummyProfile
};
