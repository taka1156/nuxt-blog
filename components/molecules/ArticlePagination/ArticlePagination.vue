<template>
  <div class="pagination">
    <base-btn :use-type="`pagenation`" @btn-click="prevPage">
      <span class="pagination__btn">
        &lt;
      </span>
    </base-btn>
    <base-text :use-type="`pagenation`" :color="`green`">
      {{ currentPage }}/{{ maxPage }}
    </base-text>
    <base-btn :use-type="`pagenation`" @btn-click="nextPage">
      <span class="pagination__btn">
        &gt;
      </span>
    </base-btn>
  </div>
</template>

<script>
import BaseBtn from '../../atoms/BaseBtn/BaseBtn';
import BaseText from '../../atoms/BaseText/BaseText';

export default {
  name: 'ArticlePagination',
  components: {
    'base-btn': BaseBtn,
    'base-text': BaseText
  },
  props: {
    /**
     * 現在閲覧しているページ番号
     */
    currentPage: {
      type: Number,
      default: 0,
      requierd: true
    },
    /**
     * 最大ページ数
     */
    maxPage: {
      type: Number,
      default: 0,
      requierd: true
    }
  },
  methods: {
    prevPage() {
      /**
       * 前のページに進む
       * (イベント伝搬)
       * @event prevPage
       */
      this.$emit('prev');
    },
    nextPage() {
      /**
       * 次のページに進む
       * (イベント伝搬)
       * @event nextPage
       */
      this.$emit('next');
    }
  }
};
</script>

<style scoped>
.pagination {
  display: flex;
  justify-content: center;
  margin: 0 auto;
  padding: 0;
  width: 100%;
  color: rgba(40, 167, 69, 0.9);
}

.pagination__btn {
  color: rgba(40, 167, 69, 0.9);
}
</style>
