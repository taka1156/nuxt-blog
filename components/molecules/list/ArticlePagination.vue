<template>
  <div class="pagination">
    <base-btn @click="$emit('prev')">
      <base-text class="base-text--green">
        &lt;
      </base-text>
    </base-btn>
    <base-text class="base-text--pagenation base-text--green">
      {{ currentPage }}/{{ maxPage }}
    </base-text>
    <base-btn @click="$emit('next')">
      <base-text class="base-text--green">
        &gt;
      </base-text>
    </base-btn>
  </div>
</template>

<script>
import BaseBtn from '../../atoms/base/BaseBtn';
import BaseText from '../../atoms/base/BaseText';

export default {
  name: 'ArticlePagenation',
  components: {
    'base-btn': BaseBtn,
    'base-text': BaseText
  },
  props: {
    currentPage: {
      type: Number,
      default: 0,
      requierd: true
    },
    maxPage: {
      type: Number,
      default: 0,
      requierd: true
    }
  }
};
</script>

<style scoped>
.pagination {
  display: flex;
  justify-content: center;
  margin: 0 auto;
  padding: 0 auto;
  width: 100%;
  color: rgba(40, 167, 69, 0.9);
}
</style>
