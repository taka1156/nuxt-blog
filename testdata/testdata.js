const dummyImg = {
  imgUrl: 'http://placehold.jp/150x150.png',
  imgAlt: 'ダミーの画像'
};

const dummyCategoryBadge = {
  badgeType: 'category',
  badge: {
    id: 1,
    name: 'ダミーカテゴリー',
    img: { url: 'http://placehold.jp/150x150.png' }
  }
};

const dummyTagBadge = {
  badgeType: 'tag',
  badge: {
    id: 1,
    name: 'ダミータグ',
    img: { url: 'http://placehold.jp/150x150.png' }
  }
};

const dummyTagBadges = [
  {
    id: 1,
    name: 'ダミータグ',
    img: { url: 'http://placehold.jp/150x150.png' }
  },
  {
    id: 2,
    name: 'ダミータグ',
    img: { url: 'http://placehold.jp/150x150.png' }
  },
  {
    id: 3,
    name: 'ダミータグ',
    img: { url: 'http://placehold.jp/150x150.png' }
  },
  {
    id: 4,
    name: 'ダミータグ',
    img: { url: 'http://placehold.jp/150x150.png' }
  }
];

const dummyDate = {
  createdAt: '2020-07-04T10:53:40.252Z',
  updatedAt: '2020-07-08T15:15:07.668Z'
};

const dummyMarkdown = `
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。** 
## この文章はダミーです。 
  **文字の大きさ、量、字間、行間等を確認するために入れています。** 
## この文章はダミーです。 
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。 
  **文字の大きさ、量、字間、行間等を確認するために入れています。** 
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。 
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。** 
## この文章はダミーです。 
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
`;

const dummyArticles = [
  {
    id: '1pbb3xp8x',
    createdAt: '2020-09-13T14:28:35.489Z',
    updatedAt: '2020-09-13T16:01:29.403Z',
    publishedAt: '2020-09-13T14:28:35.489Z',
    title:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています',
    summary:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量',
    body: dummyMarkdown,
    tags: [
      {
        id: '7zcsnuihh',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      }
    ],
    category: {
      id: 'eq7p_6jei',
      name: 'ダミー',
      img: {
        url: 'http://placehold.jp/150x150.png'
      }
    }
  },
  {
    id: 'hcpeu85l7',
    createdAt: '2020-09-13T12:38:43.014Z',
    updatedAt: '2020-09-13T15:55:21.517Z',
    publishedAt: '2020-09-13T12:38:43.014Z',
    title:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています',
    summary:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量',
    body: dummyMarkdown,
    tags: [
      {
        id: 'piymdvgeb',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      },
      {
        id: '7zcsnuihh',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      }
    ],
    category: {
      id: 'eq7p_6jei',
      name: 'ダミー',
      img: {
        url: 'http://placehold.jp/150x150.png'
      }
    }
  },
  {
    id: 'hcpeu85l7',
    createdAt: '2020-09-13T12:38:43.014Z',
    updatedAt: '2020-09-13T15:55:21.517Z',
    publishedAt: '2020-09-13T12:38:43.014Z',
    title:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています',
    summary:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量',
    body: dummyMarkdown,
    tags: [
      {
        id: 'piymdvgeb',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      },
      {
        id: '7zcsnuihh',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      }
    ],
    category: {
      id: 'eq7p_6jei',
      name: 'ダミー',
      img: {
        url: 'http://placehold.jp/150x150.png'
      }
    }
  },
  {
    id: 'hcpeu85l7',
    createdAt: '2020-09-13T12:38:43.014Z',
    updatedAt: '2020-09-13T15:55:21.517Z',
    publishedAt: '2020-09-13T12:38:43.014Z',
    title:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています',
    summary:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量',
    body: dummyMarkdown,
    tags: [
      {
        id: 'piymdvgeb',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      },
      {
        id: '7zcsnuihh',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      }
    ],
    category: {
      id: 'eq7p_6jei',
      name: 'ダミー',
      img: {
        url: 'http://placehold.jp/150x150.png'
      }
    }
  },
  {
    id: 'hcpeu85l7',
    createdAt: '2020-09-13T12:38:43.014Z',
    updatedAt: '2020-09-13T15:55:21.517Z',
    publishedAt: '2020-09-13T12:38:43.014Z',
    title:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています',
    summary:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量',
    body: dummyMarkdown,
    tags: [
      {
        id: 'piymdvgeb',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      },
      {
        id: '7zcsnuihh',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      }
    ],
    category: {
      id: 'eq7p_6jei',
      name: 'ダミー',
      img: {
        url: 'http://placehold.jp/150x150.png'
      }
    }
  },
  {
    id: 'xizlxccff',
    createdAt: '2020-09-07T04:35:28.817Z',
    updatedAt: '2020-09-07T12:59:56.511Z',
    publishedAt: '2020-09-07T04:35:28.817Z',
    title:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています',
    summary:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量',
    body: dummyMarkdown,
    tags: [
      {
        id: 'piymdvgeb',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      },
      {
        id: 'aamcmuhtf',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      },
      {
        id: 'xghqpsmpp',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      }
    ],
    category: {
      id: 'pgxhtaaa9z',
      name: 'ダミー',
      img: {
        url: 'http://placehold.jp/150x150.png'
      }
    }
  },
  {
    id: 'b-zlnpffc',
    createdAt: '2020-07-04T12:24:08.429Z',
    updatedAt: '2020-09-07T13:00:12.392Z',
    publishedAt: '2020-07-04T12:24:08.429Z',
    title:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています',
    summary:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量',
    body: dummyMarkdown,
    tags: [
      {
        id: 'svdq1mmtc',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      }
    ],
    category: {
      id: 'eq7p_6jei',
      name: 'ダミー',
      img: {
        url: 'http://placehold.jp/150x150.png'
      }
    }
  }
];

const dummyPagination = {
  currentPage: 1,
  maxPage: Math.ceil(dummyArticles.length / 5)
};

const dummyClassfication = [
  {
    id: 'xfrbr-upq',
    name: 'ダミー',
    img: {
      url: 'http://placehold.jp/150x150.png'
    }
  },
  {
    id: 'tlx_v85px',
    name: 'ダミー',
    img: {
      url: 'http://placehold.jp/150x150.png'
    }
  },
  {
    id: 'xghqpsmpp',
    name: 'ダミー',
    img: {
      url: 'http://placehold.jp/150x150.png'
    }
  },
  {
    id: 'dhyw5n7iw',
    name: 'ダミー',
    img: {
      url: 'http://placehold.jp/150x150.png'
    }
  }
];

const dummyLogo = 'ダミー';

const dummyRoutes = [
  {
    name: 'ダミー',
    to: '/test1',
    img: 'http://placehold.jp/150x150.png'
  },
  {
    name: 'ダミー',
    to: '/test2',
    img: 'http://placehold.jp/150x150.png'
  },
  {
    name: 'ダミー',
    to: '/test3',
    img: 'http://placehold.jp/150x150.png'
  },
  {
    name: 'ダミー',
    to: '/test4',
    img: 'http://placehold.jp/150x150.png'
  }
];

export {
  dummyImg,
  dummyCategoryBadge,
  dummyTagBadge,
  dummyTagBadges,
  dummyDate,
  dummyArticles,
  dummyPagination,
  dummyClassfication,
  dummyLogo,
  dummyRoutes
};
