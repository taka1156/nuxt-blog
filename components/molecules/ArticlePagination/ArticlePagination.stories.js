import ArticlePagination from './ArticlePagination.vue';
import { dummyPagination } from '@/__testdata__/testdata.js';

export default {
  title: 'Molecules/ArticlePagination',
  component: ArticlePagination
};

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { ArticlePagination },
  template: '<article-pagination v-bind="$props" />'
});

export const Default = Template.bind({});
Default.args = dummyPagination;
