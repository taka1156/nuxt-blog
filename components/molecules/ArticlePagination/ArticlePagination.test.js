import { shallowMount } from '@vue/test-utils';
import ArticlePagination from './ArticlePagination.vue';
import { dummyPagination } from '@/__testdata__/testdata.js';

describe('ArticlePagination', () => {
  const articleDate = propsData =>
    shallowMount(ArticlePagination, {
      propsData: {
        ...propsData
      }
    });

  it('ArticlePaginationの初期値: createdAt, updatedAt', () => {
    const wrapper = articleDate(dummyPagination);
    // createdAt
    expect(wrapper.vm.$options.props.currentPage.required).toBe(true);
    expect(wrapper.vm.currentPage).toBe(dummyPagination.currentPage);
    // updatedAt
    expect(wrapper.vm.$options.props.maxPage.required).toBe(true);
    expect(wrapper.vm.maxPage).toBe(dummyPagination.maxPage);
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('prevPageを実行した時、`prev`のイベントが発生するか', () => {
    const wrapper = articleDate(dummyPagination);
    wrapper.vm.prevPage();
    expect(wrapper.emitted('prev')).toBeTruthy();
  });

  it('nextPageを実行した時、`next`のイベントが発生するか', () => {
    const wrapper = articleDate(dummyPagination);
    wrapper.vm.nextPage();
    expect(wrapper.emitted('next')).toBeTruthy();
  });
});
