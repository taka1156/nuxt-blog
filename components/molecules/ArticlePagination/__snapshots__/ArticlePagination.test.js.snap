// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlePagination ArticlePaginationの初期値: createdAt, updatedAt 1`] = `
"<div>
  <div class=\\"article-pagination\\">
    <base-btn-stub><span>
        &lt;
      </span></base-btn-stub>
    <base-text-stub> 1/2 </base-text-stub>
    <base-btn-stub><span>
        &gt;
      </span></base-btn-stub>
  </div>
</div>"
`;
